docs/data/material/varnish/colors/ColorValues.tsx

```diff
@@ -0,0 +1,198 @@
+        ([colorGroupKey, colorGroup]: [string, Color[]]) => {
+          return (
+            <ColorGroup
+              key={colorGroupKey}
+              colors={colorGroup}
+              group={colorGroupKey}
+              themeProp={themeProp}
+            />
+          );
+        },
+      )}
+    </React.Fragment>
+  );
+}
+
+interface ColorGroupProps {
+  colors: Color[];
+  group: string;
+  themeProp: string;
+}
+
+function ColorGroup(props: ColorGroupProps) {
+  return (
+    <React.Fragment>
+      <ExtendedHeadRow>
+        <Label>{props.group}</Label>
+      </ExtendedHeadRow>
+      <Namespace>
+        <Typography variant="caption">{`theme.${props.themeProp}.`}</Typography>
+      </Namespace>
+      <Line />
+      {props.colors.map((col: Color) => {
+        return col ? (
+          <React.Fragment key={col.displayName}>
+            <ColorRow color={col} themeProp={props.themeProp} />
+            <Line />
+          </React.Fragment>
+        ) : null;
+      })}
+    </React.Fragment>
+  );
+}
+
+interface ColorRowProps {
+  color: Color;
+  themeProp: string;
+}
+
+function ColorRow(props: ColorRowProps) {
+  const theme = useTheme();
+  return (
+    <React.Fragment>
+      <ColorBoxWrapper>
+        <ColorBox
+          color={props.color.hex}
+          borderColor={
+            props.color.useContrastText
+              ? theme.palette.common.white
+              : theme.palette.text.primary
+          }
+        >
+          <ColorText
+            color={
+              props.color.useContrastText
+                ? theme.palette.common.white
+                : theme.palette.text.primary
+            }
+          >
+            A
+          </ColorText>
+        </ColorBox>
+      </ColorBoxWrapper>
+      <ColorName>
+        <CopyToClipboardButton
+          text={`theme.${props.themeProp}.${props.color.displayName}`}
+        >
+          {props.color.displayName}
+        </CopyToClipboardButton>
+      </ColorName>
+      <ColorHex>
+        <CopyToClipboardButton>{props.color.hex}</CopyToClipboardButton>
+      </ColorHex>
+      {props.color.rgb ? (
+        <ColorRgb>
+          <CopyToClipboardButton
+            text={`rgb(${props.color.rgb.r}, ${props.color.rgb.g}, ${props.color.rgb.b})`}
+          >
+            <span>
+              x{props.color.rgb.r}, {props.color.rgb.g}, {props.color.rgb.b}
+            </span>
+          </CopyToClipboardButton>
+        </ColorRgb>
+      ) : null}
+    </React.Fragment>
+  );
+}
+
+// TODO:  implement the <Columns count={x} /> component as seen in the Incubator project
+export const ExtendedGrid = styled('div')`
+  display: grid;
+  grid-template-columns: max-content repeat(3, auto);
+  align-items: center;
+  width: 100%;
+  grid-template-rows:
+    75px repeat(22, auto)
+    75px repeat(22, auto)
+    75px repeat(22, auto)
+    75px repeat(22, auto)
+    75px repeat(22, auto)
+    75px repeat(22, auto)
+    75px repeat(22, auto)
+    75px repeat(22, auto);
```

Comment:
Is there some sort of fancy repeat thing you can do here, to avoid repeating the row definition

Or maybe you could use [grid-auto-rows](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-auto-rows)?

Reply:
ya.. this is old code.. needs to be better